examples/i18n/RFC.md

```diff
@@ -0,0 +1,569 @@
+            "placeholder": "Enter your password"
+          },
+          "submit": "Sign in"
+        },
+        "forgot": "Forgot password?",
+        "register": "Register"
+      },
+      "register": {
+        "title": "Register.",
+        "description": "Create your account",
+        "form": {
+          "email": {
+            "label": "Email address",
+            "placeholder": "Enter your email address"
+          },
+          "password": {
+            "label": "Password",
+            "placeholder": "Enter your password"
+          },
+          "confirmPassword": {
+            "label": "Re-enter password",
+            "placeholder": "Re-enter your password"
+          },
+          "submit": "Register"
+        },
+        "login": "Login"
+      },
+      "recover": {
+        "title": "Forgot password.",
+        "description": "Enter the email address associated with your account to receive a link to reset your password",
+        "form": {
+          "email": {
+            "label": "Email address",
+            "placeholder": "Enter your email address"
+          },
+          "submit": "Reset password"
+        },
+        "login": "Login"
+      }
+    }
+  }
+}
+```
+
+#### Example `all.json`
+
+```json
+[
+  {
+    "isDefault": true,
+    "code": "US",
+    "name": "United States (USD $)",
+    "languages": [{"code": "en", "name": "English"}],
+    "prefixes": ["/en-US"]
+  },
+  {
+    "code": "CA",
+    "name": "Canada (CAD $)",
+    "languages": [
+      {"code": "en", "name": "English"},
+      {"code": "fr", "name": "French"}
+    ],
+    "prefixes": ["/en-CA", "/fr-CA"]
+  },
+  {
+    "code": "ES",
+    "name": "Spain (EUR €)",
+    "languages": [
+      {"code": "es", "name": "Spanish"},
+      {"code": "ca", "name": "Catalan"},
+      {"code": "eu", "name": "Basque"},
+      {"code": "gl", "name": "Galician"}
+    ],
+    "prefixes": ["/es-ES", "/ca-ES", "/eu-ES", "/gl-ES"]
+  },
+  {
+    "code": "FR",
+    "name": "France (EUR €)",
+    "languages": [{"code": "fr", "name": "French"}],
+    "prefixes": ["/fr-FR"]
+  },
+  {
+    "code": "DE",
+    "name": "Germany (EUR €)",
+    "languages": [{"code": "de", "name": "German"}],
+    "prefixes": ["/de-DE"]
+  }
+]
+```
+
+### 2. Create a locale parser for your given localization strategy
+
+The localization parser allows you to configure fine-grained details about your localization
+strategy. It allows you to configure the casing the country and language codes
+in the url. Here we create a `subfolder` locale parser that support urls such as:
+
+- example.com/en-CA/
+- example.com/fr-CA/
+  ....
+
+```ts
+// file: /utils.ts
+
+import {createSubfolderLocaleParser} from '@shopify/hydrogen/i18n';
+
+// Configure the i18n locale format. e.g this will match /fr-CA/ or /en-CA
+export const subfolderLocaleParser = createSubfolderLocaleParser({
+  parser: ({COUNTRY, language, delimiter}) =>
+    `/${language}${delimiter['-']}${COUNTRY}`,
```

Comment:
Would it be simpler to just make the parser be case insensitive? And the signature could also be `{country, language, delimiter}`?